steps/split.py

```diff
@@ -1,26 +1,22 @@
 """
 This module defines the following routines used by the 'split' step of the regression pipeline:
 
-- ``process_splits``: Defines customizable logic for processing & cleaning the training, validation,
+- ``mark_to_be_filtered``: Defines customizable logic for filtering the training, validation,
   and test datasets produced by the data splitting procedure. Note that arbitrary transformations
   should go into the transform step.
 """
 
-from pandas import DataFrame
+from pandas import DataFrame, Series
 
 
-def process_splits(
-    train_df: DataFrame, validation_df: DataFrame, test_df: DataFrame
-) -> (DataFrame, DataFrame, DataFrame):
+def mark_to_be_filtered(dataset: DataFrame) -> Series(bool):
```

Comment:
Was the naming of this method agreed upon by stakeholders? Seems a little bit awkward to me. I think `create_dataset_filter` is clearer.

Reply:
+1

Reply:
`mark_to_be_filtered_or_not_to_be_filtered_that_is_the_question` gets my vote

Reply:
good suggestion! changed to `create_dataset_filter`